Cargo.toml

```diff
@@ -91,7 +91,7 @@ libc = "0.2"
 libipld = { version = "0.14", default-features = false, features = ["dag-cbor", "dag-json", "derive", "serde-codec"] }
 libipld-core = { version = "0.14", features = ['serde-codec'] }
 libipld-macro = "0.14"
-libp2p = { version = "=0.51.1", default-features = false, features = [
+libp2p = { version = "0.52.1", default-features = false, features = [
```

Comment:
Pinning the shim crate here does not help pin underlying libp2p crate versions

Reply:
rust-libp2p maintainer here. You should be able to pin underlying crates via the lockfile if that is desired :)